network/p2p/p2p.go

```diff
@@ -321,3 +343,55 @@ func formatPeerTelemetryInfoProtocolName(telemetryID string, telemetryInstance s
 		base32.StdEncoding.EncodeToString([]byte(telemetryInstance)),
 	)
 }
+
+var private6 = parseCIDR([]string{
+	"100::/64",
+	"2001:2::/48",
+	"2001:db8::/32", // multiaddr v0.13 has it
+})
+
+// parseCIDR converts string CIDRs to net.IPNet.
+// function panics on errors so that it is only called during initialization.
+func parseCIDR(cidrs []string) []*net.IPNet {
+	result := make([]*net.IPNet, 0, len(cidrs))
+	var ipnet *net.IPNet
+	var err error
+	for _, cidr := range cidrs {
+		if _, ipnet, err = net.ParseCIDR(cidr); err != nil {
+			panic(err)
+		}
+		result = append(result, ipnet)
+	}
+	return result
+}
+
+// addressFilter filters out private and unroutable addresses
+func addressFilter(addrs []multiaddr.Multiaddr) []multiaddr.Multiaddr {
+	res := make([]multiaddr.Multiaddr, 0, len(addrs))
+	for _, addr := range addrs {
+		if manet.IsPublicAddr(addr) && !manet.IsPrivateAddr(addr) {
```

Comment:
What is the case where `isPublicAddr` is true as well as `isPrivateAddr`? Are we sure that in that scenario we definitely want to filter?

Reply:
You are right, `IsPublic` includes `!IsPrivate` - I'll update